core/arch/arm/plat-zynqmp/conf.mk

```diff
@@ -27,3 +27,10 @@ CFG_SHMEM_SIZE   ?= 0x10000000
 
 CFG_WITH_STATS ?= y
 CFG_CRYPTO_WITH_CE ?= y
+
+CFG_CSU_AES ?= y
+
+ifeq ($(CFG_CSU_AES), y)
+$(call force,CFG_CSU_PUF,y,Mandated by CFG_CSU_AES)
```

Comment:
In here you tie PUF to AES functionality which is not suitable for all applications -> please keep them separate as non-PUF users may want to access the AES with different key than PUF.

Reply:
I would think that image encryption feature is so important with FPGA designs so perhaps default for CSU PUF enablement would need to be `n`.

I believe we need to select what HUK solution one goes for and then that would select necessary hardware modules in use?

Reply:
right..it is a bit difficult for me to generalize on what other users might need. From my perspective - SPL authenticated boot with RPMB access, this made sense. I can work with your requirement but moving forward those with extended needs can as well build on top? I just dont like to generalize too early unless there is a need (like in this case)

Reply:
Well you could reverse it with something like this:
```makefile
CFG_CSU_PUF ?= y

ifeq ($(CFG_CSU_PUF), y)
$(call force,CFG_CSU_AES,y,Needed by CFG_CSU_PUF)
$(call force,CFG_CSU_DMA,y,Needed by CFG_CSU_PUF)
endif
```

Then later you might have something like:

```makefile
CFG_XILINX_MPSOC_HUK_PUF ?= y

ifeq ($(CFG_XILINX_MPSOC_HUK_PUF), y)
$(call force,CFG_CSU_PUF,y,Needed by CFG_XILINX_MPSOC_HUK_PUF)
endif
```